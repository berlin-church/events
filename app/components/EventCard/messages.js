/*
 * EventCard Messages
 *
 * This contains all the text for the EventCard component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.EventCard';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'This is the EventCard component!',
  },
  connect: {
    id: `${scope}.connect`,
    defaultMessage: 'CONNECT',
  },
  hostCommunity: {
    id: `${scope}.hostCommunity`,
    defaultMessage: 'HOST A COMMUNITY GROUP',
  },
  commnityGroup: {
    id: `${scope}.commnityGroup`,
    defaultMessage: 'COMMUNITY GROUP',
  },
});
