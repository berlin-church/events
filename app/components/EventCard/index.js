/**
 *
 * EventCard
 *
 */

import PropTypes from 'prop-types';

import React, { memo } from 'react';
import { FormattedMessage } from 'react-intl';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Button from '@material-ui/core/Button';

import messages from './messages';

const useStyles = makeStyles({
  card: {
    margin: '0.75rem',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: 'min-content',
  },
  media: {
    position: 'relative',
  },
  link: {
    color: 'orange',
    fontSize: '1.2em',
    borderTop: '1px solid rgba(160,160,160,0.2)',
    width: '100%',
    justifyContent: 'flex-start',
    height: '3em',
  },
  title: {
    position: 'absolute',
    top: '70%',
    left: '10%',
    color: 'white',
    fontSize: '1.5em',
  },
});

function EventCard(props) {
  const { event } = props;

  const formattedMsgObj = messages[event.msgKey];

  const classes = useStyles();
  return (
    <Card className={classes.card}>
      <CardActionArea>
        <CardMedia
          component="img"
          className={classes.media}
          image={event.image}
          title={event.title}
        />
        <div className={classes.title}>
          <FormattedMessage {...formattedMsgObj} />
        </div>
        <CardContent />
      </CardActionArea>
      <CardActions>
        <Button size="small" className={classes.link}>
          {event.url}
        </Button>
      </CardActions>
    </Card>
  );
}

EventCard.propTypes = {
  event: PropTypes.object,
};

export default memo(EventCard);
