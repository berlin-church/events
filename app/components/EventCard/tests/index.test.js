/**
 *
 * Tests for EventCard
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import EventCard from '../index';
import { DEFAULT_LOCALE } from '../../../i18n';

describe('<EventCard />', () => {
  it('Should render and match the snapshot', () => {
    const event = {
      title: 'Eu',
      image:
        'https://ichef.bbci.co.uk/news/660/cpsprodpb/9C76/production/_105745004_gettyimages-684415304.jpg',
      url:
        'https://ichef.bbci.co.uk/news/660/cpsprodpb/9C76/production/_105745004_gettyimages-684415304.jpg',
      msgKey: 'connect',
    };

    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <EventCard event={event} />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
