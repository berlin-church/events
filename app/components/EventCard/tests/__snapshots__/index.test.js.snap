// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventCard /> Should render and match the snapshot 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded MuiCard-root makeStyles-card-1"
>
  <button
    class="MuiButtonBase-root MuiCardActionArea-root"
    tabindex="0"
    type="button"
  >
    <img
      class="MuiCardMedia-root MuiCardMedia-media makeStyles-media-2"
      src="https://ichef.bbci.co.uk/news/660/cpsprodpb/9C76/production/_105745004_gettyimages-684415304.jpg"
      title="Eu"
    />
    <div
      class="makeStyles-title-4"
    >
      <span>
        CONNECT
      </span>
    </div>
    <div
      class="MuiCardContent-root"
    />
    <span
      class="MuiCardActionArea-focusHighlight"
    />
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <div
    class="MuiCardActions-root MuiCardActions-spacing"
  >
    <button
      class="MuiButtonBase-root MuiButton-root makeStyles-link-3 MuiButton-text MuiButton-sizeSmall"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        https://ichef.bbci.co.uk/news/660/cpsprodpb/9C76/production/_105745004_gettyimages-684415304.jpg
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
